File: SparkDemo/src/main/scala/org/abondar/experimental/sparkdemo/scala/StreamFromConsole.scala
package org.abondar.experimental.sparkdemo.scala

import org.apache.spark.SparkConf
import org.apache.spark.streaming.{Seconds, StreamingContext}


object StreamFromConsole {

  def main(args: Array[String]): Unit = {
    val conf = new SparkConf().setAppName("Stream from console")
    val ssc = new StreamingContext(conf,Seconds(2))

    val lines = ssc.socketTextStream("172.17.0.1", 7777)
    val errorLines = lines.filter(_.contains("error"))
    errorLines.print()

    ssc.start()
    ssc.awaitTermination()
    ssc.stop()
  }

}

File: SparkDemo/src/main/java/org/abondar/experimental/sparkdemo/java/StreamFromConsole.java
package org.abondar.experimental.sparkdemo.java;


import org.apache.spark.SparkConf;
import org.apache.spark.api.java.function.Function;
import org.apache.spark.streaming.Durations;
import org.apache.spark.streaming.api.java.JavaDStream;
import org.apache.spark.streaming.api.java.JavaStreamingContext;

public class StreamFromConsole {

    public static void main(String[] args) throws InterruptedException {
        SparkConf conf = new SparkConf();
        JavaStreamingContext jssc = new JavaStreamingContext(conf, Durations.seconds(2));

        JavaDStream<String> lines = jssc.socketTextStream("localhost", 7777);
        JavaDStream<String> errorLines = lines.filter((Function<String, Boolean>) line -> line.contains("error"));
        errorLines.print();

        jssc.start();
        jssc.awaitTermination();
        jssc.stop();
    }
}
